p11-kit/test-transport3.c

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright (c) 2012 Stefan Walter
+ * Copyright (c) 2012-2023 Red Hat Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above
+ *       copyright notice, this list of conditions and the
+ *       following disclaimer.
+ *     * Redistributions in binary form must reproduce the
+ *       above copyright notice, this list of conditions and
+ *       the following disclaimer in the documentation and/or
+ *       other materials provided with the distribution.
+ *     * The names of contributors to this software may not be
+ *       used to endorse or promote products derived from this
+ *       software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * Authors: Stef Walter <stef@thewalter.net>
+ *          Jakub Jelen <jjelen@redhat.com>
+ */
+
+#include "config.h"
+#include "test.h"
+
+#include "library.h"
+#include "mock.h"
+#include "path.h"
+#include "private.h"
+
+#include "p11-kit.h"
+#include "rpc.h"
+
+#include <errno.h>
+#include <sys/types.h>
+#include <signal.h>
+#ifdef OS_UNIX
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#endif
+#include <stdlib.h>
+#include <stdio.h>
+
+struct {
+	char *directory;
+	char *user_config;
+	char *user_modules;
+#ifdef OS_UNIX
+	pid_t pid;
+#endif
+} test;
+
+static void
+setup_remote (void *unused)
```

Comment:
This test also shares quite a lot with `p11-kit/test-transport.c`. Can these two be consolidated as well?

Reply:
ok

Reply:
I think there is a bit of a problem with this.
In test-rpc I was able to merge them into one because I used v3 module but tested both v2 and v3 interfaces.
However in test-transport only the used module changes. So either I keep it separate to test v2 module and v3 module or I merge it such that I only test v3 module. I dont know how else I would do it.